synapse/storage/databases/main/search.py

```diff
@@ -834,7 +835,7 @@ def _tokenize_query(query: str) -> TokenList:
         in_phrase = bool(i % 2)
 
         # Pull out the individual words, discarding any non-word characters.
-        words = deque(re.findall(r"([\w\-]+)", part, re.UNICODE))
+        words = deque(parse_words(part))
```

Comment:
Note that this is only used when using sqlite, we don't do any tokenizing when using postgresql.